### PR TITLE
Sillitski/hometask_4

### PR DESCRIPTION
**Pull request by qqqwww123cdv**

### CODE REVIEW

File: qqqwww123cdv/hometask-4/my-array-rotate/my_array_rotate_benchmark.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+require 'benchmark'
+require './my_array_rotate'
+
+Benchmark.bm do |x|
+  p 'my lab_rotate'
+  x.report { (1..100).to_a.lab_rotate(10) }
+  x.report { (1..100).to_a.lab_rotate(10) }
+  x.report { (1..100).to_a.lab_rotate(10) }
+  x.report { (1..100).to_a.lab_rotate(10) }
+  x.report { (1..100).to_a.lab_rotate(10) }
```
Comment (jekapulpit) [inline qqqwww123cdv/hometask-4/my-array-rotate/my_array_rotate_benchmark.rb]: let's use array size increasingly (e.g. from 10 to 10000) to have more decent statistics

File: qqqwww123cdv/hometask-4/my-hash/my_hash.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  def []=(key, value)
+    @hash.push([key, value])
+  end
+
+  def clear
+    @hash = reject { |value| value }
+  end
+
+  def delete(key)
+    @hash = reject { |value| value.include? key }
```
Comment (jekapulpit) [inline qqqwww123cdv/hometask-4/my-hash/my_hash.rb]: `Hash#delete` returns deleted value

File: qqqwww123cdv/hometask-4/my-hash/my_hash.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  def [](key)
+    myhash[key]
+  end
+
+  def []=(key, value)
+    @hash.push([key, value])
+  end
+
+  def clear
+    @hash = reject { |value| value }
```
Comment (jekapulpit) [inline qqqwww123cdv/hometask-4/my-hash/my_hash.rb]: we need just to pass a truthy value in `reject`, so using the argument is redundant here

File: qqqwww123cdv/hometask-4/my-hash/my_hash_benchmark.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+# frozen_string_literal: true
+
+require 'benchmark'
+require './my_hash'
+
+Benchmark.bm do |x|
+  puts 'MyHash'
+  my_hash = MyHash.new
+  x.report { 100.times { |value| my_hash[value] = value } }
+  x.report { 100.times { |value| my_hash[value] = value } }
```
Comment (jekapulpit) [inline qqqwww123cdv/hometask-4/my-hash/my_hash_benchmark.rb]: also here, let's use values increasingly

**Review by jekapulpit — CHANGES_REQUESTED** [review]

**Comment by qqqwww123cdv** [comment]
image

**Review by jekapulpit — APPROVED** [review]

**Pull request merged**